controller/src/util/pod_watcher.rs

```diff
@@ -317,16 +331,18 @@ impl BrokerPodWatcher {
         )
         .await?;
 
-        // Make sure instance has required Pods
-        if let Ok(instance) = kube_interface.find_instance(&instance_id, namespace).await {
-            super::instance_action::handle_instance_change(
-                &instance,
-                &super::instance_action::InstanceAction::Update,
-                kube_interface,
-            )
-            .await?;
+        // Only redeploy Pods that are managed by the Akri Controller (have Instance ownerReference)
```

Comment:
i'm a little confused here ... what happens here if !instance_owned?  does get_instance_and_configuration_from_pod return successfully if the pod is not instance_owned?  what does it return in that case?

Reply:
Yes it returns successfully. For terminating pods, this call to handle_instance_change makes sure that pods are re-deployed if terminated. For jobs, that check isnt necessary and would hit [this logging line](https://github.com/kate-goldenring/akri/blob/strategy-jobs/controller/src/util/instance_action.rs#L478) and do nothing. This basically stops unnecessary code from running. It assumes that if an instance resource is being use by an owner other than Akri's controller (ie a Job) then we don't have further action to do when a pod stops running.

Reply:
ok, so BrokerJobSpec pods aren't owned ... but have akri bits in the set of labels.  that wasn't clear to me before.

Reply:
Yes all pods are owned by the Job. But we own the job

Reply:
"we own the job" ... does that mean the instance owns the job?  and the job owns the pod?

Reply:
yes exactly. sorry for not being more specific

Reply:
i think i might change some of the variable names and functions to try to clarify that somehow.

maybe `fn instance_owned -> bool` can be `fn get_broker_pod_type_from_ownership -> BrokerPodType` where BrokerPodType is an enum: [BrokerPod, BrokerJob, Unknown]

Reply:
I see your point. One thing that linguistically confuses me is how Jobs are a management object that deploy Pods. So in the Pod Watcher, we are reacting to Pods being created by the Job. Is it fair to call that Pod a BrokerJob?

Reply:
Maybe just `get_broker_type_from_ownership` (removing pod) would help

Reply:
`get_broker_type_from_ownership` works for me

Reply:
for me, this will help clarify the if statements ... the intent/meaning of `if type == BrokerPod {...}` is much clearer to me than `if instance_owned {...}`